### PR TITLE
Neurotoxin rebalance, new chemical Ozemolyn, purges chemical and does minor toxin damage to capped amount.

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
Rebalances neurotoxin to no longer purge, no other changes.

1.5u metab rate, 3 purge rate. 0.75 tox damage up to 40 points.

Adds new chemical to xenomorphs, Ozemolyn. Defiler gets it in gaseous form, queen/shrike/drone in sting form and removed neurotoxin from them inplace of this. Works exactly like neurosting but with a different chem.

Defiler gas is non transparent like neurogas, which should give it some nice utility compared to other forms of gas.

Makes neurotoxin sting a bit more generic, it now injects whatever chemical is specified in the variable.

## Why It's Good For The Game
Adds a new chem seperating neurotoxin from one of it's points and allowing for better balance.

This is better in the long run and balances neuros absurd rates a bit, it still does pain and stamina damage.

## Changelog
:cl:
add: Added new xenomorph chemical, Ozemolyn, purges chemical and does minor toxin damage to capped amount. Sting for Queen/Shrike/Drone. Gaseous form for Defiler, gas is transparent.
balance: neurotoxin no longer purges. Removed neuro sting from shrike, queen and drone.
/:cl:

### CODE REVIEW

**Comment by private-tristan** [comment]
seems cool

File: code/datums/keybinding/xeno.dm [hunk 1]
```diff
@@ -94,6 +94,12 @@
 	description = "A channeled melee attack that injects the target with neurotoxin over a few seconds, temporarily stunning them."
 	keybind_signal = COMSIG_XENOABILITY_NEUROTOX_STING
 
+/datum/keybinding/xeno/ozelomelyn_sting
+	name = "ozelomelyn_sting"
+	full_name = "Ozelomelyn Sting"
+	description = "A channeled melee attack that injects the target with Ozelomelyn over a few seconds, purging chemicals and dealing minor toxin damage to a moderate cap while inside them."
+	keybind_signal = COMSIG_XENOABILITY_OZELOMELYN_STING
```
Comment (BraveMole) [inline code/datums/keybinding/xeno.dm:101]: this should probably use the same keybind as regular sting? I won't force this, if you think it's better to have two keybinds then ok

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 2]
```diff
@@ -908,6 +908,56 @@
...
+	var/mob/living/carbon/xenomorph/X = owner
+
+	succeed_activate()
+
+	add_cooldown()
+
+	GLOB.round_statistics.ozelomelyn_stings++
+	SSblackbox.record_feedback("tally", "round_statistics", 1, "ozelomelyn_stings")
+
+	X.recurring_injection(A, /datum/reagent/toxin/xeno_ozelomelyn, XENO_NEURO_CHANNEL_TIME, XENO_NEURO_AMOUNT_RECURRING)
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:959]: no, this should be a child of the other sting

File: code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm [hunk 3]
```diff
@@ -1,9 +1,10 @@
 #define DEFILER_NEUROTOXIN "Neurotoxin"
 #define DEFILER_HEMODILE "Hemodile"
 #define DEFILER_TRANSVITOX "Transvitox"
+#define DEFILER_OZELOMELYN "Ozelomelyn"
 
 GLOBAL_LIST_INIT(defile_purge_list, typecacheof(list(
-	/datum/reagent/toxin/xeno_hemodile, /datum/reagent/toxin/xeno_transvitox, /datum/reagent/toxin/xeno_neurotoxin)))
+	/datum/reagent/toxin/xeno_ozelomelyn, /datum/reagent/toxin/xeno_hemodile, /datum/reagent/toxin/xeno_transvitox, /datum/reagent/toxin/xeno_neurotoxin)))
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm:7]: i dislike having 4 chems for defiler, when most players don't know what the heck are the three others

File: code/modules/reagents/reagents/toxin.dm [hunk 4]
```diff
@@ -649,6 +649,29 @@
...
+	color = "#f1ddcf"
+	custom_metabolism = 1.5 // metabolizes decently quickly. A sting does 15 at the same rate as neurotoxin.
+	overdose_threshold = 10000
+	scannable = TRUE
+	toxpwr = 0 // This is going to do slightly snowflake tox damage.
+	purge_list = list(/datum/reagent/medicine)
+	purge_rate = 5
+
+/datum/reagent/toxin/xeno_ozelomelyn/on_mob_life(mob/living/L, metabolism)
+	var/tox_loss = L.getToxLoss() // get the mobs toxin damage
```
Comment (BraveMole) [inline code/modules/reagents/reagents/toxin.dm:665]: this var is useless

File: code/modules/reagents/reagents/toxin.dm [hunk 5]
```diff
@@ -649,6 +649,29 @@
...
+	if(tox_loss < 40) // if our toxloss is below 40, do 0.75 tox damage.
+		L.adjustToxLoss(0.75)
+		if(prob(15))
+			to_chat(L, span_warning("Your veins feel like water and you can feel a growing itchy feeling in them!") )
+	else
```
Comment (BraveMole) [inline code/modules/reagents/reagents/toxin.dm:670]: needs an early return

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Comment by Hecklingheck** [comment]
I think xenos, or at the very least the toxin-oriented castes like Defiler, should have a way of knowing what chemicals/medicines and how much of said medicines are inside a marine.
Otherwise, using this new Ozemolyn chemical would be too clunky, and often counterproductive since it also purges neurotox and such.  Information is pretty important

File: code/datums/keybinding/xeno.dm (hunk 1 above)
Comment (Pariah919) [reply code/datums/keybinding/xeno.dm:101]: better to have two keybinds incase we have to give it to sentinel or whatever.

File: code/modules/mob/living/carbon/xenomorph/abilities.dm [hunk 6]
```diff
@@ -869,6 +869,7 @@
 	keybind_signal = COMSIG_XENOABILITY_NEUROTOX_STING
 	target_flags = XABB_MOB_TARGET
 	use_state_flags = XACT_USE_BUCKLED
+	var/sting_chemical = /datum/reagent/toxin/xeno_neurotoxin /// Whatever our victim is injected with.
```
Comment (BraveMole) [inline code/modules/mob/living/carbon/xenomorph/abilities.dm:872]:
```suggestion
	/// Whatever our victim is injected with.
	var/sting_chemical = /datum/reagent/toxin/xeno_neurotoxin 
```

File: code/modules/reagents/reagents/toxin.dm [hunk 7]
```diff
@@ -649,6 +649,29 @@
...
+	color = "#f1ddcf"
+	custom_metabolism = 1.5 // metabolizes decently quickly. A sting does 15 at the same rate as neurotoxin.
+	overdose_threshold = 10000
+	scannable = TRUE
+	toxpwr = 0 // This is going to do slightly snowflake tox damage.
+	purge_list = list(/datum/reagent/medicine)
+	purge_rate = 5
+
+/datum/reagent/toxin/xeno_ozelomelyn/on_mob_life(mob/living/L, metabolism)
+	L.getToxLoss() // get the mobs toxin damage
```
Comment (BraveMole) [inline code/modules/reagents/reagents/toxin.dm:665]: this line is useless

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/modules/reagents/reagents/toxin.dm [hunk 8]
```diff
@@ -649,6 +649,28 @@
...
+	toxpwr = 0 // This is going to do slightly snowflake tox damage.
+	purge_list = list(/datum/reagent/medicine)
+	purge_rate = 5
+
+/datum/reagent/toxin/xeno_ozelomelyn/on_mob_life(mob/living/L, metabolism)
+	if(L.getToxLoss() < 40) // if our toxloss is below 40, do 0.75 tox damage.
+		L.adjustToxLoss(0.75)
+		if(prob(15))
+			to_chat(L, span_warning("Your veins feel like water and you can feel a growing itchy feeling in them!") )
+			return ..()
```
Comment (BraveMole) [inline code/modules/reagents/reagents/toxin.dm:669]:
```suggestion
		return ..()
```

**Review by BraveMole — CHANGES_REQUESTED** [review]

**Review by BraveMole — APPROVED** [review]

**Comment by Mechanelize** [comment]
does this make hemodile improvement work with the new chem or still with neuro and transvitox

**Pull request merged**